File: adapter/python/myadapter.py

class FoodProcessor:
    def process(self, food_item):
        return f"Processing {food_item} in myadapter."


class Chopper:
    def chop(self, food_item):
        return f"Chopping {food_item} in myadapter."

class ChopperAdapter(FoodProcessor, Chopper):
    def process(self, food_item):
        return self.chop(food_item)

if __name__ == "__main__":
    fp = FoodProcessor()
    fp.process("carrot")
    
    chopper = Chopper()
    chopper.chop("onion")
    
    chopper_adapter = ChopperAdapter()
    chopper_adapter.process("carrot")
    chopper_adapter.chop("onion")

File: adapter/swift/myadapter.swift
protocol Chopper {
    func chop() -> String
}

class FoodChopper: Chopper {
    func chop() -> String {
        return "Choppping Food."
    } 
}

protocol Processor {
    func processFood() -> String
}

class FoodProcessor: Processor {
    func processFood() -> String {
        return "Processing Food"
    }
}

struct NewFoodProcessor: Chopper {
    var processor = FoodProcessor()

    init(processor: FoodProcessor) {
        self.processor = processor
    }

    func chop() -> String {
        return "New Food Processor chopping food"
    }
}

@main
struct MyAdapter {
    static func main() {
        let foodProcessor = FoodProcessor()
        print(foodProcessor.processFood())

        let chopper = NewFoodProcessor(processor: foodProcessor)
        print(chopper.chop())
    }
}

// swiftc -parse-as-library myadapter.swift
// ./myadapter